failing fuzz test - WalWriterFuzzTest.testAddDropColumnDropPartition()

### To reproduce

Run `WalWriterFuzzTest.testAddDropColumnDropPartition()` with these seeds: 
```
        s0 = 872206675813L;
        s1 = 1738242160678L;
```

QuestDB commit: 42ebd0a69f923c4
The test is failing consistently with:
```
java.lang.AssertionError: file descriptors, expected: [4294967296,103079215107,98784247813,107374182408], actual: [4294967296,103079215107,98784247813,472446404878,476741372175,481036339472,523986012433,107374193863] 
Expected :3
Actual   :7
<Click to see difference>
	at junit@4.13.2/org.junit.Assert.fail(Assert.java:89)
	at junit@4.13.2/org.junit.Assert.failNotEquals(Assert.java:835)
	at junit@4.13.2/org.junit.Assert.assertEquals(Assert.java:647)
	at io.questdb.test/io.questdb.test.tools.TestUtils$LeakCheck.close(TestUtils.java:1999)
	at io.questdb.test/io.questdb.test.tools.TestUtils.assertMemoryLeak(TestUtils.java:670)
	at io.questdb.test/io.questdb.test.AbstractCairoTest.assertMemoryLeak(AbstractCairoTest.java:1061)
	at io.questdb.test/io.questdb.test.cairo.fuzz.AbstractFuzzTest.runFuzz(AbstractFuzzTest.java:191)
	at io.questdb.test/io.questdb.test.cairo.fuzz.WalWriterFuzzTest.runFuzz(WalWriterFuzzTest.java:557)
	at io.questdb.test/io.questdb.test.cairo.fuzz.WalWriterFuzzTest.testAddDropColumnDropPartition(WalWriterFuzzTest.java:100)
[...]
```

### QuestDB version:

8.2.2

### OS, in case of Docker specify Docker and the Host OS:

Linux@Intel

### File System, in case of Docker specify Host File System:

ext4

### Full Name:

Jaromir Hamala

### Affiliation:

QuestDB

### Have you followed Linux, MacOs kernel configuration steps to increase Maximum open files and Maximum virtual memory areas limit?

- [x] Yes, I have

### Additional context

_No response_

--- a/core/src/main/java/io/questdb/cairo/TableWriter.java
+++ b/core/src/main/java/io/questdb/cairo/TableWriter.java
@@ -4234,11 +4234,12 @@ public class TableWriter implements TableWriterAPI, MetadataService, Closeable {
                 denseSymbolMapWriters.size(),
                 txWriter
         );
+        denseSymbolMapWriters.add(w);
+        symbolMapWriters.extendAndSet(columnCount, w);
+        
         // In case there are some dirty files left from rolled back transaction
         // clean the newly created symbol files.
         w.truncate();
-        denseSymbolMapWriters.add(w);
-        symbolMapWriters.extendAndSet(columnCount, w);
     }
 
     private boolean createWalSymbolMapping(SymbolMapDiff symbolMapDiff, int columnIndex, IntList symbolMap) {

--- a/core/src/main/java/io/questdb/cairo/TableWriter.java
+++ b/core/src/main/java/io/questdb/cairo/TableWriter.java
@@ -4234,12 +4234,20 @@ public class TableWriter implements TableWriterAPI, MetadataService, Closeable {
                 denseSymbolMapWriters.size(),
                 txWriter
         );
+
+        try {
+            // In case there are some dirty files left from rolled back transaction
+            // clean the newly created symbol files.
+            w.truncate();
+        } catch (Throwable t) {
+            // oh, well, we tried and it failed. this can happen if there is e.g. I/O issue.
+            // we can't do much about it but make sure we close the writer to avoid leaks
+            w.close();
+            throw t;
+        }
+
         denseSymbolMapWriters.add(w);
         symbolMapWriters.extendAndSet(columnCount, w);
-
-        // In case there are some dirty files left from rolled back transaction
-        // clean the newly created symbol files.
-        w.truncate();
     }
 
     private boolean createWalSymbolMapping(SymbolMapDiff symbolMapDiff, int columnIndex, IntList symbolMap) {

--- a/core/src/main/java/io/questdb/cairo/TableWriter.java
+++ b/core/src/main/java/io/questdb/cairo/TableWriter.java
@@ -4236,7 +4236,7 @@ public class TableWriter implements TableWriterAPI, MetadataService, Closeable {
         );
         denseSymbolMapWriters.add(w);
         symbolMapWriters.extendAndSet(columnCount, w);
-        
+
         // In case there are some dirty files left from rolled back transaction
         // clean the newly created symbol files.
         w.truncate();

